### PR TITLE
Adapt the API of the joint embedding task

### PR DESCRIPTION
**Pull request by mumichae**
Add organism and double-check that mandatory keys are specified accordingly.

### CODE REVIEW

**Comment by rcannood** [comment]
Thanks @mumichae ! Would you be able to also update the API descriptions for the common dataset loaders at `src/common/datasets/README.md`? :bow:

**Comment by mumichae** [comment]
I added some more keys for cell cycle and trajectory inference scores and marked them as optional. I'd need those entries for the test sets so that cell cycle and TI conservation scores work on the test set.

File: src/joint_embedding/README.md [hunk 1]
```diff
@@ -31,19 +38,27 @@ This component should output *three* h5ad files, `--output_mod1`, `--output_mod2
...
+  * `.X`: Sparse profile matrix of `"GEX"` features only.
+  * `.uns['dataset_id']`: Name of the dataset.
+  * `.obs['batch']`: Batch id.
+  * `.obs['cell_type']`: Cell type each cell belongs to.
+  * `.obs['organism']`: Organism the cell was taken from.
+  * `.obs['S_score']`: Cell cycle score on S-phase genes (optional unless `var_names` aren't gene symbols).
+  * `.obs['G2M_score']`: Cell cycle score on G2-phase & M-phase genes (optional unless `var_names` aren't gene symbols).
+  * `.obs['RNA_trajectory']`: Trajectory scores for `"GEX"` features (optional).
+  * `.obs['ATAC_trajectory']`: Trajectory scores for `"ATAC"` features (optional).
+  * `.obs['ADT_trajectory']`: Trajectory scores for `"ADT"` features (optional).
```
Comment (LuckyMD) [inline src/joint_embedding/README.md:59]: Instead of trajectory scores, let's say "pseudtome values"

File: src/joint_embedding/README.md [hunk 2]
```diff
@@ -31,19 +38,27 @@ This component should output *three* h5ad files, `--output_mod1`, `--output_mod2
...
   * `.obs_names`: Ids for the cells.
+  * `.var_names`: Ids for the features.
```
Comment (LuckyMD) [inline src/joint_embedding/README.md:61]: specify GEX features

**Review by LuckyMD — COMMENTED** [review]
General concepts are good, maybe just change some of the wording.

File: src/common/datasets/README.md [hunk 3]
```diff
@@ -16,6 +16,12 @@ These AnnData files should both have the following attributes:
   * `ad.var['feature_types']`: The modality of this feature. For `output_rna`, this should be equal to `"GEX"`. For `output_mod2`, this should be equal to `"ATAC"` or `"ADT"` depending on the dataset.
   * `ad.obs['batch']`: A batch identifier (optional). If available, this can be used downstream to make train/test splits.
   * `ad.obs['cell_type']`: A cell type (optional). If available, this dataset can be used for task 2, otherwise not.
+  * `.obs['organism']`: Organism the cell was taken from (optional).
+  * `.obs['S_score']`: Cell cycle score on S-phase genes only for `.var['feature_types'] == 'GEX'` (optional).
+  * `.obs['G2M_score']`: Cell cycle score on G2-phase & M-phase genes (optional).
+  * `.obs['RNA_trajectory']`: Trajectory scores for `"GEX"` features (optional).
+  * `.obs['ATAC_trajectory']`: Trajectory scores for `"ATAC"` features (optional).
+  * `.obs['ADT_trajectory']`: Trajectory scores for `"ADT"` features (optional).
```
[the remark this answers is not in the document]
Comment (mumichae) [reply src/common/datasets/README.md:24]: Good idea! Do batch and cell type also belong to optional output or should we require them?

File: src/common/datasets/README.md [hunk 4]
```diff
@@ -11,19 +11,25 @@ A DataGen component does not have any required inputs. Take a look at each compo
...
+  * `.obs_names`: Ids for the cells.
+  * `.var_names`: Ids for the features.
+
+### Optional Attributes
+
+For certain tasks, additional attributes are needed.
+The AnnData files should contain:
+
+  * `.obs['batch']`: A batch identifier. If available, this can be used downstream to make train/test splits.
+  * `.obs['cell_type']`: A cell type. If available, this dataset can be used for task 2, otherwise not.
```
Comment (LuckyMD) [inline src/common/datasets/README.md:26]: task 2 is now task 3 => joint embedding task

**Review by LuckyMD — APPROVED** [review]
only 1 small thing about task 2/task 3 naming: maybe just call it the joint embedding task. Otherwise good from my side. @rcannood I guess this will depend on if you can generate some G2M and S phase scores in the dyngen data...

Also, we might need data loader tests for some of these things.

**Comment by rcannood** [comment]
The suggested changes for the API look good! 

For generating these outputs with dyngen, I have an idea on how to implement values for `batch` and `cell_type`.

For `S_score` and `G2M_score`, I assume these will scale between [0, 1]?

For the `RNA_trajectory`, the `ATAC_trajectory` and the `ADT_trajectory`: how do you define these values? Are you assuming that the trajectory is linear? How do you compute these values; run a trajectory inference method on it?

**Comment by LuckyMD** [comment]
Hey! 

The trajectory scores are the output of `sc.tl.dpt()` on a subset of cells relevant to the trajectory and a root cell chosen as a starting point. These are just pseudotime values. The S and G2M scores I'm not sure are scaled to [0,1]... they can also be negative as far as I'm aware. They are generated by `sc.tl.score_genes_cell_cycle()`.

**Pull request merged**